text/0001-poet2-consensus.md

```diff
@@ -0,0 +1,109 @@
+5. Register the ECDSA Public key and the IAS response with the ledger. 
+6. Wait for ‘C’ blocks to be added to the chain before participating in consensus.
+
+## Wall Clock and Chain Clock
+The PoET 2.0 algorithm maintains two clocks, the _Wall Clock_ and the _Chain Clock_.
+The Wall Clock (WC) measures the time elapsed since the arrival of the ‘Sync Block’, i.e. the first block on the chain.
+The Chain Clock (CC) measures the cumulative wait duration of all the blocks in the chain (each fork will have its own CC).
+Upon receiving the Sync Block, WC and CC are initialized to 0. In practice, the WC may be calculated by recording the system time at the moment of the arrival of the Sync Block and subsequently subtracting this timestamp from the current time.
+To handle clock drifts, it may become necessary to synchronize with a standard network clock. Details are out of scope for this version of the document.
+
+## Block Creation
+Once the algorithm is in a state to publish blocks (i.e. the C test passes), it starts the process of creating claim blocks with the aim of participating in the Consensus mechanism.
+The claim/candidate block is assembled as usual with no special processing required for PoET 2.0
+
+## Wait Certificate Creation
+Once a claim block is created, the PoET algorithm requests the PoET enclave to create the WaitCertificate.
+
+The contents of the WaitCertificate for PoET 2.0 are:
+Duration: A random 256 bit number generated by SGX
+WaitTime: The number of seconds to wait, determined by the LocalMean
+BlockID: The BlockID passed in to the Enclave
+PrevBlockID: The BlockID of the previous block, as stored in the previousWaitCert
+TxnHash: The hash of the transactions in the block
+BlockNumber: The length of the chain
+ValidatorID: The ID of the current Validator
+Sign: The signature of the WaitCertificate computed over all the fields using the ECDSA private key
+
+Once the WaitCertificate is created, the enclave stores the Last Block # and the WaitCertificate in the table created previously. The validator will only create a single wait certificate per block. This implies that if a fork containing fewer blocks becomes active, the validator will NOT be able to publish any more blocks until the new fork has added enough blocks to catch up to the previous chain.
+
+## Block Publishing
+With the ClaimBlock ready and the WaitCertificate created, the validator will determine the duration for which to set the timer. The duration of the wait is determined by the ‘Duration’ field in the WaitCertificate.
+Ideally, any selected algorithm should be capable of converting the duration to a wait time distributed uniformly between 0 and LocalMean seconds.
+
+## Block Verification
+Upon receiving a claim block and a WaitCertificate, the receiving validator does the following:
+### Sanity checks
+1.	Ensure the Block and WaitCertificate signatures are valid
+2.	The Block’s ID matches the WaitCert’s blockID
```

Comment:
Another nitpick: I suggest spelling out "WaitCertificate" (here and elsewhere) instead of abbreviating it in a few places. Consistency is good. :-)